tests/mock/readme.md

```diff
@@ -0,0 +1,194 @@
+# Snowpark Local Testing
+
+Snowpark Local Testing allows the usage of creating DataFrames and
+performing operations without a connection to Snowflake.
+
+## Quickstart
+
+Instead of using `Session.SessionBuilder` to create a session, in local testing,
+instantiate `Session` with a `MockServerConnection` object.
+
+```python
+from snowflake.snowpark.session import Session
+from snowflake.snowpark.mock.connection import MockServerConnection
+session = Session(MockServerConnection())
+df = session.create_dataframe(
+    [
+        [1, 2, "welcome"],
+        [3, 4, "to"],
+        [5, 6, "the"],
+        [7, 8, "private"],
+        [9, 0, "preview"]
+    ],
+    schema=['a', 'b', 'c']
+)
+df.select('c').show()
+```
+
+## General Usage Documentation
+
+### Installation
+
+The Snowpark local testing framework can be installed from the [development branch](https://github.com/snowflakedb/snowpark-python/tree/dev/local-testing) on the public repository. In your requirements.txt, add the following:
+
+```bash
+pip install "snowflake-snowpark-python[pandas]@git+https://github.com/snowflakedb/snowpark-python@dev/local-testing"
+```
+
+#### Install from specific commit
+
+The installation instructions above will install Snowpark from the development branch, meaning that any new commits pushed to that branch will be installed to your Python environment the next time you install the package. You can also set the installation to a specific commit using the syntax below:
+
+```bash
+pip install "snowflake-snowpark-python[pandas]@git+https://github.com/snowflakedb/snowpark-python@dev/local-testing@29385014c755fe20122fac536358a8ebdeb761bc"
+```
+
+If you use this approach, you will need to manually update the commit if you want to use features that are added to the branch in future commits.
+
+
+## Patching Built-In Functions
+
+Not all the built-in functions under `snowflake.snowpark.functions` have been re-implemented for the local testing framework.
```

Comment:
I guess users don't understan what is "patching built-in functions"? Needs to explain it first.